core/src/androidMain/kotlin/com/ustadmobile/core/catalog/contenttype/VideoTypePluginAndroid.kt

```diff
@@ -7,162 +7,275 @@ import android.media.MediaFormat
+                        setInteger(MediaFormat.KEY_BIT_RATE, AUDIO_BIT_RATE)
                     }
 
-                    override fun onCancelled(id: String, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
-                        Napier.d(tag = VIDEO_ANDROID, message = "cancelled transform")
-                        videoCompleted.complete(false)
+                    val videoCompleted = CompletableDeferred<Boolean>()
+
+                    val mediaTransformer = MediaTransformer(context as Context)
+                    mediaTransformer.transform(contentJobItem.cjiContentEntryUid.toString(), localUri.uri, newVideo.path,
+                            videoTarget, audioTarget, object : TransformationListener {
+                        override fun onStarted(id: String) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "started transform")
+                        }
+
+                        override fun onProgress(id: String, progress: Float) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "progress at value ${progress * 100}")
+                            contentJobItem.cjiItemProgress = ((progress * contentJobItem.cjiItemTotal) / progressSize).toLong()
+                            jobProgress.onProgress(contentJobItem)
+                        }
+
+                        override fun onCompleted(id: String, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "completed transform")
+                            contentJobItem.cjiItemProgress = contentJobItem.cjiItemTotal / progressSize
+                            jobProgress.onProgress(contentJobItem)
+                            videoCompleted.complete(true)
+                        }
+
+                        override fun onCancelled(id: String, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "cancelled transform")
+                            videoCompleted.complete(false)
+                        }
+
+                        override fun onError(id: String, cause: Throwable?, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            videoCompleted.completeExceptionally(cause
+                                    ?: RuntimeException("error on video id: $id"))
+                        }
+
+                    }, TransformationOptions.Builder()
+                            .setGranularity(MediaTransformer.GRANULARITY_DEFAULT)
+                            .setVideoFilters(null)
+                            .build()
+                    )
+
+
+                    try {
+                        videoCompleted.await()
+                    } catch (e: Exception) {
+                        Napier.e(tag = VIDEO_ANDROID, throwable = e, message = e.message ?: "")
```

Comment:
I think if there is an exception here, we need to stop. Why continue creating and uploading a container if the conversion has failed? It should throw a fatal exception (which the runner will catch)